src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Models/ProvideSemanticTokensResponse.cs

```diff
@@ -9,19 +9,38 @@ namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic
 {
     internal class ProvideSemanticTokensResponse
     {
-        public ProvideSemanticTokensResponse(SemanticTokens result, long? hostDocumentSyncVersion)
+        public ProvideSemanticTokensResponse(SemanticTokens? result, SemanticTokensDelta? editResult, long? hostDocumentSyncVersion)
```

Comment:
To me it makes more sense to make two separate constructors, but when I tried it, I received some json serialization exceptions that only one constructor is allowed.

Reply:
Does it make sense to have different types to represent a delta result vs. a full result? @ryanbrandenburg would know better than I

Reply:
The "classical" way of doing this would be to use a `SumType<SemanticTokens, SemanticTokensDelta>` but I think there was some trouble there, where did that land?